chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb

```diff
@@ -0,0 +1,53 @@
+#
+# Cookbook Name:: crowbar
+# Recipe:: crowbar-upgrade
+#
+# Copyright 2013-2015, SUSE LINUX Products GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# TODO check for node state: if it is ready, we're likely returning
+# from previous upgrade state and need to re-enable chef-client
+# otherwise do the disabling stuff bellow:
+
+# Disable openstack services
+# We don't know which openstack services are enabled on the node and
+# collecting that information via the attributes provided by chef is
+# is rather complicated. So instead we fall back to a simple bash hack
+
```

Comment:
Because we now have this also on the runlist for some non-upgrade states we need to add a check here for the node state on only do the disable steps when state is crowbar-upgrade. Some like:

```
case node["state"]
  when "crowbar-upgrade"
    do stuff
  when "ready","readying","applying"
    do other stuff
end
```